### PR TITLE
Add provider meta module_name in Equinix Metal TF configs

### PR DESCRIPTION
**Pull request by codinja1188**
What this PR does / why we need it:
It allows Vendor to declare the metadata fields.  It adds  provider meta module_name in Equinix Metal TF configs

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):

Related to https://github.com/equinix/terraform-provider-equinix/issues/252

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -3,9 +3,9 @@ terraform {
     null = {
       source = "hashicorp/null"
     }
-    metal = {
-      source = "equinix/metal"
-      version = "2.1.0" 
+    equinix = {
+      source = "equinix/equinix"
+      version = "1.14.1"
```
Comment (ctreatma) [inline versions.tf]: I think we should leave our [version constraints more open](https://developer.hashicorp.com/terraform/language/expressions/version-constraints#version-constraint-syntax); for example, `~> 1.14`, which would make it easier for users to keep up with the latest non-breaking changes.

File: variables.tf [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
 
-variable metal_organization_id {
+variable equinix_organization_id {
   type        = string
   default     = "null"
   description = "Equinix Metal Organization ID"
 }
 
-variable "metal_device_reservations" {
+variable "equinix__device_reservations" {
```
Comment (ctreatma) [inline variables.tf]: There are 2 underscores after `equinix` here; could you remove one?

File: variables.tf [hunk 3]
```diff
@@ -42,7 +42,7 @@ variable "metal_device_reservations" {
   type        = map(any)
   default     = {}
 }
-variable metal_device_metro {
+variable equinix__device_metro {
```
Comment (ctreatma) [inline variables.tf]: There are 2 underscores after `equinix` here; could you remove one?

File: equinix_metal.tf [hunk 4]
```diff
@@ -3,7 +3,7 @@ resource "tls_private_key" "ssh_key_pair" {
   rsa_bits  = 4096
 }
 
-resource "metal_ssh_key" "ssh_pub_key" {
+resource "equinix_ssh_key" "ssh_pub_key" {
```
Comment (ctreatma) [inline equinix_metal.tf]: This should be [`equinix_metal_ssh_key`](https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_ssh_key)

File: equinix_metal.tf [hunk 5]
```diff
@@ -21,44 +21,44 @@ data "template_file" "user_data" {
   }
 }
 
-resource "metal_project" "new_project" {
-  count           = var.metal_create_project ? 1 : 0
+resource "equinix_project" "new_project" {
```
Comment (ctreatma) [inline equinix_metal.tf]: This should be [`equinix_metal_project`](https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_project)

File: equinix_metal.tf [hunk 6]
```diff
@@ -21,44 +21,44 @@ data "template_file" "user_data" {
...
+  count           = var.equinix_create_project ? 1 : 0
 
-  name            = var.metal_project_name
-  organization_id = var.metal_organization_id
+  name            = var.equinix_project_name
+  organization_id = var.equinix_organization_id
 }
 
-resource "metal_device" "control_plane" {
+resource "equinix_device" "control_plane" {
```
Comment (ctreatma) [inline equinix_metal.tf]: This should be [`equinix_metal_device`](https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_device)

**Review by ctreatma — CHANGES_REQUESTED** [review]
A couple updated resource names do not match the name of the resource in the equinix provider, which [causes the CI to fail](https://github.com/equinix/terraform-equinix-metal-ibm-satellite-on-baremetal/actions/runs/4917617923/jobs/8782939739?pr=7).  If you have an Equinix Metal account, it may be helpful to run the updated module locally using an API token from your account.

File: versions.tf (hunk 1 above)
Comment (codinja1188) [reply versions.tf]: done

**Review by ctreatma — APPROVED** [review]

**Pull request merged**